nemo_aligner/models/nlp/gpt/megatron_gpt_dpo_model.py

```diff
@@ -128,7 +128,12 @@ def fwd_output_and_loss_func(dataloader_iter, model, checkpoint_activations_all_
             if batch["chosen_labels"] is not None and batch["rejected_labels"] is not None:
                 labels = torch.cat((batch["chosen_labels"], batch["rejected_labels"]), dim=0)
 
-            if batch["ref_policy_log_probs_chosen"] is not None and batch["ref_policy_log_probs_rejected"] is not None:
+            if (
+                "ref_policy_log_probs_chosen" in batch
+                and "ref_policy_log_probs_rejected" in batch
+                and batch["ref_policy_log_probs_chosen"] is not None
+                and batch["ref_policy_log_probs_rejected"] is not None
```

Comment:
Can be written more succinctly:
```suggestion
                batch.get("ref_policy_log_probs_chosen") is not None
                and batch.get("ref_policy_log_probs_rejected") is not None
```

Reply:
> Instead of removing log_prob_forward_micro_batch_size, can we set it to default ${multiply: 2, ${..micro_batch_size}} with a comment that says using a different value may cause a logprobs mismatch vs. the reference policy? (if that's the problem)

This does not seem to work. @gshennvm  and I have been setting `log_prob_forward_micro_batch_size = 2` with `micro_batch_size=1`. The issue persists.


> In l.184 we have
> ```
> 
>                 per_token_logps = from_parallel_logits_to_logprobs(
>                     vocab_parallel_logits=output_tensor, target=labels, higher_stability=True
>                 )
> ```
> but shouldn't it use inference_only=validation_step? (this is not new in this PR, but I just noticed it)

Seems to me that you are right.

Reply:
> This does not seem to work. @gshennvm and I have been setting log_prob_forward_micro_batch_size = 2 with micro_batch_size=1. The issue persists.

To be clear, I'm not saying this is the only thing to change. My point is that your new code is hardcoding
```python
micro_batch_size=self.cfg.micro_batch_size * 2
```
but I believe it'd be better to make this the default behavior while allowing one to change it if they want to (this also avoids a breaking change).